### DISCUSSION TITLE
How to Access Variable Values at Specific Points in a Kernel

### DISCUSSION THREAD

**Discussion by hityyds** [idaholab/moose#27861] [Q&A General]
### Check these boxes if you have followed the posting rules.

- [X] Q&A General is the most appropriate section for my question
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (screenshots as a last resort, triple back quotes around pasted text)

### Question

I want to compute a 2D equation that includes a term representing the difference in a variable's values at two positions, such as `u(x,1)−u(x,−1)`. How can I access the variable solution at arbitrary points in a Kernel?

Thank you all for your attention. I would be especially grateful if someone could provide a solution to this issue.

**Comment by GiudGiud** [comment]
Hello

The easiest way to do this will be to use the Functor arguments of variables.
You use a ElemPointArg as an argument to the evaluation of the variable, with something similar to this pseudo code (use the doxygen to find the actual routines):

```
// First, get a point locator from the mesh
_pl = _mesh.getPointLocator();
const auto other_elem = _pl.locate_elem(point + (0,1,0));
const ElemPointArg elem_point(other_elem, point + (0,1,0));
const auto value_other_elem = _var(elem_point, determineState());
```

then this gets you the location at the a point + 1 on the second axis

Guillaume
